Maelabord4.rmd

```diff
@@ -0,0 +1,728 @@
+---
+title: "Mælaborð"
+author: "The North"
+date: "2024-06-11"
+output: html_document
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+```{r}
+# Hlaða nauðsynlegum bókasöfnum
+library(shiny)
+library(dplyr)
+library(tidyr)
+library(ggplot2)
+library(DBI)
+library(RSQLite)
+library(stringr)
+library(bslib)
+library(plotly)
+
+# Tengist SQLite gagnagrunninum
+conn <- dbConnect(SQLite(), "/Users/asdishalla/Desktop/capstone-the-north-Gagnagrunnur 2/data/rotten_tomatoes.db")
+
+
+# Les öll gögnin úr gagnagrunninum
+rotten_tomatoes_data <- dbReadTable(conn, "rotten_tomatoes_movies")
+rotten_tomatoes_leo_kate <- dbReadTable(conn, "rotten_tomatoes_movies_dicaprio_winslet")
+leo_movies <- dbReadTable(conn, "leo_movies")
+kate_movies <- dbReadTable(conn, "kate_movies")
+main_actors_in_info <- dbReadTable(conn, "main_actors_in_info")
+content_rating_count <- dbReadTable(conn, "content_rating_count")
+movie_actors <- dbReadTable(conn, "movie_actors")
+actors_info <- dbReadTable(conn, "actors_info")
+
+
+actors_info <- dbGetQuery(conn, "
+  SELECT headshot, actor_name, mini_biography, trivia_1, trivia_2, highest_paid_movie, highest_salary
+  FROM actors_info
+")
+
```

Comment:
vantar þetta hérna en skiptir ekki máli er buin að bæta því við hjá mer: ### Query for Top 10 Tomatometer-Rated Movies
query_top10 <- "
SELECT movie_title, tomatometer_rating, audience_rating, directors, actors, original_release_date
FROM rotten_tomatoes_movies
WHERE actors LIKE '%Kate Winslet%' OR actors LIKE '%Leonardo DiCaprio%'
ORDER BY tomatometer_rating DESC
LIMIT 10
"
top10_movies <- dbGetQuery(conn, query_top10)
# Extract unique directors for Tomatometer
directors <- unique(top10_movies$directors)
### Query for All Movies by These Directors
query_all_movies <- sprintf("
SELECT movie_title, tomatometer_rating, audience_rating, directors, actors, original_release_date
FROM rotten_tomatoes_movies
WHERE directors IN (%s)
", paste(sprintf("'%s'", directors), collapse = ", "))
all_movies <- dbGetQuery(conn, query_all_movies)
### Query for Top 10 Audience-Rated Movies
query_top10_audience <- "
SELECT movie_title, tomatometer_rating, audience_rating, directors, actors, original_release_date
FROM rotten_tomatoes_movies
WHERE actors LIKE '%Kate Winslet%' OR actors LIKE '%Leonardo DiCaprio%'
ORDER BY audience_rating DESC
LIMIT 10
"
top10_audience_movies <- dbGetQuery(conn, query_top10_audience)
# Extract unique directors for Audience
directors_audience <- unique(top10_audience_movies$directors)
### Query for All Movies by These Directors
query_all_audience_movies <- sprintf("
SELECT movie_title, tomatometer_rating, audience_rating, directors, actors, original_release_date
FROM rotten_tomatoes_movies
WHERE directors IN (%s)
", paste(sprintf("'%s'", directors_audience), collapse = ", "))
all_audience_movies <- dbGetQuery(conn, query_all_audience_movies)
### Cleanup: Disconnect from the Database